accord-core/src/main/java/accord/local/PreLoadContext.java

```diff
@@ -38,37 +39,47 @@
     /**
      * @return keys of the {@link CommandsForKey} objects that need to be loaded into memory before this operation is run
      */
-    Iterable<Key> keys();
+    Seekables<?, ?> keys();
 
-    static PreLoadContext contextFor(Iterable<TxnId> txnIds, Iterable<Key> keys)
+    static PreLoadContext contextFor(Iterable<TxnId> txnIds, Seekables<?, ?> keys)
     {
         return new PreLoadContext()
         {
             @Override
             public Iterable<TxnId> txnIds() { return txnIds; }
 
             @Override
-            public Iterable<Key> keys() { return keys; }
+            public Seekables<?, ?> keys() { return keys; }
         };
     }
 
-    static PreLoadContext contextFor(TxnId txnId, Iterable<Key> keys)
+    static PreLoadContext contextFor(TxnId txnId, Seekables<?, ?> keysOrRanges)
     {
-        return contextFor(Collections.singleton(txnId), keys);
+        switch (keysOrRanges.kindOfContents())
+        {
+            default: throw new AssertionError();
+            case Range: return contextFor(txnId); // TODO (soon): this won't work for actual range queries
```

Comment:
Why?

Reply:
I guess this gets into weird stuff like is Range over Routables or Seekables. Currently it's really just over Routables. I guess it doesn't matter since what is here can't be used to support seekable ranges yet.